tests/parametric/test_otel_metrics.py

```diff
@@ -0,0 +1,2445 @@
+                name,
+                DEFAULT_INSTRUMENT_UNIT,
+                DEFAULT_INSTRUMENT_DESCRIPTION,
+                non_negative_value,
+                DEFAULT_MEASUREMENT_ATTRIBUTES,
+            )
+            t.otel_counter_add(
+                DEFAULT_METER_NAME,
+                name,
+                DEFAULT_INSTRUMENT_UNIT,
+                DEFAULT_INSTRUMENT_DESCRIPTION,
+                second_non_negative_value,
+                NON_DEFAULT_MEASUREMENT_ATTRIBUTES,
+            )
+            t.otel_metrics_force_flush()
+
+        metrics = test_agent.wait_for_num_otlp_metrics(num=1)
+        scope_metrics = metrics[0]["resource_metrics"][0]["scope_metrics"]
+
+        # Assert that the ScopeMetrics has the correct Scope, SchemaUrl, and Metrics data
+        assert_scope_metric(
+            scope_metrics[0], DEFAULT_METER_NAME, DEFAULT_METER_VERSION, DEFAULT_SCHEMA_URL, DEFAULT_SCOPE_ATTRIBUTES
+        )
+
+        metric = scope_metrics[0]["metrics"][0]
+        assert_metric_info(metric, name, DEFAULT_INSTRUMENT_UNIT, DEFAULT_INSTRUMENT_DESCRIPTION)
+        assert_sum_aggregation(
+            metric["sum"],
+            "AGGREGATION_TEMPORALITY_DELTA",
+            is_monotonic=True,
+            value=non_negative_value,
+            attributes=DEFAULT_MEASUREMENT_ATTRIBUTES,
+        )
+        assert_sum_aggregation(
+            metric["sum"],
+            "AGGREGATION_TEMPORALITY_DELTA",
+            is_monotonic=True,
+            value=second_non_negative_value,
+            attributes=NON_DEFAULT_MEASUREMENT_ATTRIBUTES,
+        )
+
+    # This test takes upwards of 25 seconds to run
+    @pytest.mark.parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    @given(
+        st.integers(min_value=-(2**32), max_value=2**32)
+    )  # Limit the range of integers to avoid int/float equality issues
+    @settings(
+        suppress_health_check=[HealthCheck.function_scoped_fixture], deadline=None, max_examples=20
+    )  # Limit the number of examples to speed up the test
+    def test_otel_updowncounter_add_value(self, test_agent, test_library, n):
+        name = f"updowncounter1-{n}"
+
+        with test_library as t:
+            t.otel_get_meter(DEFAULT_METER_NAME, DEFAULT_METER_VERSION, DEFAULT_SCHEMA_URL, DEFAULT_SCOPE_ATTRIBUTES)
+            t.otel_metrics_force_flush()
+            t.otel_create_updowncounter(
+                DEFAULT_METER_NAME, name, DEFAULT_INSTRUMENT_UNIT, DEFAULT_INSTRUMENT_DESCRIPTION
+            )
+            t.otel_updowncounter_add(
+                DEFAULT_METER_NAME,
+                name,
+                DEFAULT_INSTRUMENT_UNIT,
+                DEFAULT_INSTRUMENT_DESCRIPTION,
+                n,
+                DEFAULT_MEASUREMENT_ATTRIBUTES,
+            )
+            t.otel_metrics_force_flush()
+
+        metrics = test_agent.wait_for_num_otlp_metrics(num=1)
+        scope_metrics = metrics[0]["resource_metrics"][0]["scope_metrics"]
```

Comment:
Can we include a comment on why we need multiple force flushes in this test. Testing with one flush could reduce flakiness and race conditions